### PR TITLE
Update content of pft_params.nml options under User Guide

### PR DESCRIPTION
**Pull request by rkutteh**
## Description

Updated content of pft_params.nml options under User Guide as a side task of another GW-related task.
Added text and tables to provide more and neater content.

## Type of change

- [ ] New or updated documentation

----
📚 Documentation preview 📚: https://cable--299.org.readthedocs.build/en/299/

### CODE REVIEW

File: documentation/docs/user_guide/inputs/pft_params_nml.md [hunk 1]
```diff
@@ -1,5 +1,162 @@
...
+|    17	 |  Ice 		|
+
+The CABLE vegetation parameters are listed in Table 2 along with their value ranges, units, and descriptions. All parameters in Table 2 are of type real and dimension 17, corresponding to the number of vegetation types in Table 1.
+The ranges listed are “physically possible” and mostly correspond to the optional checks in the CABLE offline code.
+## Table 2: CABLE vegetation parameters
+
+| Name (vegin%)         | Range            | Units                                                                                               | Description |
+|-----------------------|------------------|-----------------------------------------------------------------------------------------------------|-------------|                                                                      
+| 	a1gs            | >=0.0        	   | \( (-) \) 												 | a1 parameter in stomatal conductance model. Represents the sensitivity of stomatal conductance to the assimilation rate |
+| 	alpha           | >=0.0        	   | \( (mol (electrons) \cdot mol^{-1} (photons) (C3) \cdot mol (CO2) \cdot mol^{-1} (photons) (C4)) \) | Initial slope of J-Q response curve |
```
Comment (ccarouge) [inline documentation/docs/user_guide/inputs/pft_params_nml.md:37]: Markdown will not split a word in the middle and considers the maths line as 1 word. So I suggest to add a manual break to improve the rendering:
```suggestion
| 	alpha           | >=0.0        	   | \( (mol (electrons) \cdot mol^{-1} (photons) (C3) \cdot \) <br> \( mol (CO2) \cdot mol^{-1} (photons) (C4)) \) | Initial slope of J-Q response curve |
```

**Review by ccarouge — CHANGES_REQUESTED** [review]
I like the new table for the default values. Some minor changes proposed, in particular to improve the rendering of table 2.

File: documentation/docs/user_guide/inputs/pft_params_nml.md (hunk 1 above)
Comment (rkutteh) [reply documentation/docs/user_guide/inputs/pft_params_nml.md:37]: I am happy with the suggestion to solve the wide column issue. Do you want me to click "Commit suggestion" or you prefer to make the change yourself ?

**Review by ccarouge — APPROVED** [review]
Good to go. You are the one to merge it in main when you are certain it is ready.

**Pull request merged**